### PR TITLE
event json support and some tweaks

### PR DESCRIPTION
**Pull request by xadhoom**

### CODE REVIEW

File: lib/elixir_mod_event/connection.ex [hunk 1]
```diff
@@ -371,6 +371,9 @@ defmodule FSModEvent.Connection do
 
   def handle_info({:tcp_closed, _}, state) do
     Logger.info "Connection closed"
+    Enum.each state.listeners, fn({_, v}) ->
+      send v.pid, {:stop, :normal, state}
```
Comment (marcelog) [inline lib/elixir_mod_event/connection.ex:375]: How about prepending the message with `fs_` and adding the pid so the listener can be sure that this is a message sent by an fs connection and also from a _specific_ fs connection? Also, the `state` is internal to the connection and should not be known to the listeners (i.e the message should be: `{:fs_close, self}`)

Comment (marcelog) [inline lib/elixir_mod_event/connection.ex:375]: By the way, I'm thinking that this is not needed at all (the message) if one just `start_link`s or monitor the connection from an OTP process (i.e: a gen_server, etc). So we can just as well remove the message.

File: lib/elixir_mod_event/connection.ex [hunk 2]
```diff
@@ -380,8 +383,11 @@ defmodule FSModEvent.Connection do
   end
 
   @spec terminate(term, FSModEvent.Connection.t) :: :ok
-  def terminate(reason, _state) do
+  def terminate(reason, state) do
     Logger.info "Terminating with #{inspect reason}"
+    Enum.each state.listeners, fn({_, v}) ->
+      send v.pid, {:terminate, reason}
```
Comment (marcelog) [inline lib/elixir_mod_event/connection.ex:389]: Same as above.. either we send the pid or we assume the connection is going to be `start_link`ed from a process and/or monitored as well

**Comment by marcelog** [comment]
Hey @xadhoom :)

Thanks for the contribution!

It looks good overall, and I'm sorry to bother you: but is it possible to close and split this pull into different pull requests according to what each of them do? Applying different patches in different pulls (for different things) will help in tracing changes and reviewing the code.

Thanks in advance!

**Comment by xadhoom** [comment]
sure, going to close and reopen as requested. I'll remove the messages for stop, since it makes no sense after all.

**Pull request closed without merging**